Reply/app/src/main/java/com/materialstudies/reply/ui/compose/ComposeFragment.kt

```diff
@@ -106,8 +124,73 @@ class ComposeFragment : Fragment() {
                 false
             ).apply {
                 account = acnt
+                root.setOnClickListener {
+                    // Bind the views in the expanded card view to this account's details when
+                    // clicked and expand.
+                    binding.focusedRecipient = acnt
+                    expandChip(it)
+                }
             }
             addView(chipBinding.root)
         }
     }
+
+    /**
+     * Expand the recipient [chip] into a popup with a list of contact addresses to choose from.
+     */
+    private fun expandChip(chip: View) {
+        val transform = MaterialContainerTransform().apply {
+            startView = chip
+            endView = binding.recipientCardView
+            scrimColor = Color.TRANSPARENT
+            // Have the transform match the endView card's native elevation as closely as possible.
+            endElevation = requireContext().resources.getDimension(
+                R.dimen.email_recipient_card_popup_elevation_compat
+            )
+            // Avoid having this transform from running on both the start and end views by setting
+            // its target to the endView.
+            addTarget(binding.recipientCardView)
+        }
+
+        // Configure the analogous collapse transform back to the recipient chip. This should
+        // happen when the card is clicked, any region outside of the card (the card's transparent
+        // scrim) is clicked, or when the back button is pressed.
+        binding.run {
+            recipientCardView.setOnClickListener { collapseChip(chip) }
+            recipientCardScrim.visibility = View.VISIBLE
+            recipientCardScrim.setOnClickListener { collapseChip(chip) }
+        }
+        closeRecipientCardOnBackPressed.expandedChip = chip
+        closeRecipientCardOnBackPressed.isEnabled = true
```

Comment:
since instantiating the transform above doesn't really have an effect yet anyway, how about we move this binding.run + closeRecipientCardOnBackPressed block to the beginning of this method? that way all of the transform code is in one place